src/ResourceManager/Resources/Commands.Resources/ActiveDirectory/NewAzureADAppCredentialCommand.cs

```diff
@@ -72,19 +73,16 @@ public override void ExecuteCmdlet()
                     ObjectId = ActiveDirectoryClient.GetObjectIdFromApplicationId(ApplicationId);
                 }
 
-#pragma warning disable 0618
-                if (!string.IsNullOrEmpty(Password))
-#pragma warning restore 0618
+                string decodedPassword = SecureStringExtensions.ConvertToString(Password);
```

Comment:
This will throw an error if Password is null, please decode once you've checked if Password is null or empty.